src/msg/async/ProtocolV2.cc

```diff
@@ -1419,6 +1434,12 @@ CtPtr ProtocolV2::handle_message() {
   message->set_throttle_stamp(throttle_stamp);
   message->set_recv_complete_stamp(ceph_clock_now());
 
+  connection->logger->tinc(l_msgr_all_read_lat,
+                           ceph_clock_now() - recv_stamp);
+  log_latency(__func__,
```

Comment:
Hmm, looks you want to log messages experiencing a processing time (counted from reading the preamble, through throttling, ending on the first stage of decoding) above a configurable threshold.

```cpp
CtPtr ProtocolV2::handle_read_frame_preamble_main(rx_buffer_t &&buffer, int r) {
  // ...
  // does it need throttle?
  if (next_tag == Tag::MESSAGE) {
    if (state != READY) {
      lderr(cct) << __func__ << " not in ready state!" << dendl;
      return _fault();
    }
    recv_stamp = ceph_clock_now();
    state = THROTTLE_MESSAGE;
    return CONTINUE(throttle_message);
  } else {
    return read_frame_segment();
  }
}
```

However, is it worth introducing a dedicated machinery with its own config option if we already have `OpTracker`? Maybe this bit should be a part of it?

Reply:
I added a configurable threshold, I wrote it with reference to `bluestore_log_op_age`，because it will report latancy to log file while the disk is slow. So, I want to add it to msgr.